### ISSUE TITLE
Custom fan curve deactivates XG mobile fan

### ISSUE THREAD

**Issue by Plumbio** [seerge/g-helper#881]
**Describe the bug**
When I enable a custom fan curve or switch to a power plan with a custom curve already activated and then connect the XG mobile, the fan of the xg mobile won't activate, and will stay that way until the gpu powers itself off because of overheating.
If I enable the fan curve after the xg mobile is connected, it's fan speed will lock to that current speed regardless the temperature.

**Clear scenario to Reproduce**
1. Enable custom fan curve or switch to a power plan with a custom curve already enabled.
2. Connect and switch to XG Mobile.

**Expected behavior**
The app already has the option to set a custom fan curve for the xg mobile, and I think I remember using it without any problems before. Maybe something in a recent update caused the issue.

**App Logs**
Timestamp                                 : Mon Jul 24 17:20:35 2023
Driver Version                            : 535.98
CUDA Version                              : 12.2

Attached GPUs                             : 1
GPU 00000000:01:00.0
    Product Name                          : NVIDIA GeForce GTX 1650 with Max-Q Design
    Product Brand                         : GeForce
    Product Architecture                  : Turing
    Display Mode                          : Disabled
    Display Active                        : Disabled
    Persistence Mode                      : N/A
    Addressing Mode                       : N/A
    MIG Mode
        Current                           : N/A
        Pending                           : N/A
    Accounting Mode                       : Disabled
    Accounting Mode Buffer Size           : 4000
    Driver Model
        Current                           : WDDM
        Pending                           : WDDM
    Serial Number                         : N/A
    GPU UUID                              : GPU-ecbea748-e47e-d6e3-524c-07806f963a20
    Minor Number                          : N/A
    VBIOS Version                         : 90.17.77.00.0d
    MultiGPU Board                        : No
    Board ID                              : 0x100
    Board Part Number                     : N/A
    GPU Part Number                       : 1F9D-755-A1
    FRU Part Number                       : N/A
    Module ID                             : 1
    Inforom Version
        Image Version                     : G001.0000.02.04
        OEM Object                        : 1.1
        ECC Object                        : N/A
        Power Management Object           : N/A
    GPU Operation Mode
        Current                           : N/A
        Pending                           : N/A
    GSP Firmware Version                  : N/A
    GPU Virtualization Mode
        Virtualization Mode               : None
        Host VGPU Mode                    : N/A
    GPU Reset Status
        Reset Required                    : No
        Drain and Reset Recommended       : N/A
    IBMNPU
        Relaxed Ordering Mode             : N/A
    PCI
        Bus                               : 0x01
        Device                            : 0x00
        Domain                            : 0x0000
        Device Id                         : 0x1F9D10DE
        Bus Id                            : 00000000:01:00.0
        Sub System Id                     : 0x149C1043
        GPU Link Info
            PCIe Generation
                Max                       : 3
                Current                   : 3
                Device Current            : 3
                Device Max                : 4
                Host Max                  : 3
            Link Width
                Max                       : 16x
                Current                   : 8x
        Bridge Chip
            Type                          : N/A
            Firmware                      : N/A
        Replays Since Reset               : 0
        Replay Number Rollovers           : 0
        Tx Throughput                     : 1000 KB/s
        Rx Throughput                     : 39000 KB/s
        Atomic Caps Inbound               : N/A
        Atomic Caps Outbound              : N/A
    Fan Speed                             : N/A
    Performance State                     : P0
    Clocks Event Reasons
        Idle                              : Not Active
        Applications Clocks Setting       : Not Active
        SW Power Cap                      : Not Active
        HW Slowdown                       : Not Active
            HW Thermal Slowdown           : Not Active
            HW Power Brake Slowdown       : Not Active
        Sync Boost                        : Not Active
        SW Thermal Slowdown               : Not Active
        Display Clock Setting             : Not Active
    FB Memory Usage
        Total                             : 4096 MiB
        Reserved                          : 147 MiB
        Used                              : 44 MiB
        Free                              : 3904 MiB
    BAR1 Memory Usage
        Total                             : 256 MiB
        Used                              : 2 MiB
        Free                              : 254 MiB
    Conf Compute Protected Memory Usage
        Total                             : N/A
        Used                              : N/A
        Free                              : N/A
    Compute Mode                          : Default
    Utilization
        Gpu                               : 0 %
        Memory                            : 0 %
        Encoder                           : 0 %
        Decoder                           : 0 %
        JPEG                              : 0 %
        OFA                               : 0 %
    Encoder Stats
        Active Sessions                   : 0
        Average FPS                       : 0
        Average Latency                   : 0
    FBC Stats
        Active Sessions                   : 0
        Average FPS                       : 0
        Average Latency                   : 0
    ECC Mode
        Current                           : N/A
        Pending                           : N/A
    ECC Errors
        Volatile
            SRAM Correctable              : N/A
            SRAM Uncorrectable            : N/A
            DRAM Correctable              : N/A
            DRAM Uncorrectable            : N/A
        Aggregate
            SRAM Correctable              : N/A
            SRAM Uncorrectable            : N/A
            DRAM Correctable              : N/A
            DRAM Uncorrectable            : N/A
    Retired Pages
        Single Bit ECC                    : N/A
        Double Bit ECC                    : N/A
        Pending Page Blacklist            : N/A
    Remapped Rows                         : N/A
    Temperature
        GPU Current Temp                  : 33 C
        GPU T.Limit Temp                  : N/A
        GPU Shutdown Temp                 : 99 C
        GPU Slowdown Temp                 : 94 C
        GPU Max Operating Temp            : 75 C
        GPU Target Temperature            : N/A
        Memory Current Temp               : N/A
        Memory Max Operating Temp         : N/A
    GPU Power Readings
        Power Draw                        : 14.02 W
        Current Power Limit               : 30.00 W
        Requested Power Limit             : 35.00 W
        Default Power Limit               : 35.00 W
        Min Power Limit                   : 1.00 W
        Max Power Limit                   : 40.00 W
    Module Power Readings
        Power Draw                        : N/A
        Current Power Limit               : N/A
        Requested Power Limit             : N/A
        Default Power Limit               : N/A
        Min Power Limit                   : N/A
        Max Power Limit                   : N/A
    Clocks
        Graphics                          : 990 MHz
        SM                                : 990 MHz
        Memory                            : 6000 MHz
        Video                             : 960 MHz
    Applications Clocks
        Graphics                          : N/A
        Memory                            : N/A
    Default Applications Clocks
        Graphics                          : N/A
        Memory                            : N/A
    Deferred Clocks
        Memory                            : N/A
    Max Clocks
        Graphics                          : 1740 MHz
        SM                                : 1740 MHz
        Memory                            : 5001 MHz
        Video                             : 1605 MHz
    Max Customer Boost Clocks
        Graphics                          : 1740 MHz
    Clock Policy
        Auto Boost                        : N/A
        Auto Boost Default                : N/A
    Voltage
        Graphics                          : N/A
    Fabric
        State                             : N/A
        Status                            : N/A
    Processes
        GPU instance ID                   : N/A
        Compute instance ID               : N/A
        Process ID                        : 1744
            Type                          : C+G
            Name                          : C:\Windows\System32\dwm.exe
            Used GPU Memory               : Not available in WDDM driver model
        GPU instance ID                   : N/A
        Compute instance ID               : N/A
        Process ID                        : 8224
            Type                          : C+G
            Name                          : C:\Windows\SystemApps\Microsoft.LockApp_cw5n1h2txyewy\LockApp.exe
            Used GPU Memory               : Not available in WDDM driver model
        GPU instance ID                   : N/A
        Compute instance ID               : N/A
        Process ID                        : 8592
            Type                          : C+G
            Name                          : C:\Windows\SystemApps\MicrosoftWindows.Client.CBS_cw5n1h2txyewy\SearchHost.exe
            Used GPU Memory               : Not available in WDDM driver model
        GPU instance ID                   : N/A
        Compute instance ID               : N/A
        Process ID                        : 8616
            Type                          : C+G
            Name                          : C:\Windows\SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\StartMenuExperienceHost.exe
            Used GPU Memory               : Not available in WDDM driver model
        GPU instance ID                   : N/A
        Compute instance ID               : N/A
        Process ID                        : 11780
            Type                          : C
            Name                          : C:\Windows\System32\clinfo.exe
            Used GPU Memory               : Not available in WDDM driver model

24-07-2023 5:20:36: Stopped: RadeonSoftware
24-07-2023 5:20:36: Running eco command 1
24-07-2023 5:20:36: Display configuration changed.
24-07-2023 5:20:36: Changed Lid Action to 1
24-07-2023 5:20:36: Disengaging Clamshell Mode
24-07-2023 5:20:36: GPUEco = 1 : OK
24-07-2023 5:20:36: PowerLimit A3 = 11 : OK
24-07-2023 5:20:36: Eco flag : 1
24-07-2023 5:20:36: Mux flag : -65536
24-07-2023 5:20:36: LQ134N1JW52:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
24-07-2023 5:35:21: Key: 56
24-07-2023 5:35:22: Latest version 0.100.0.0
24-07-2023 5:35:24: Key: 56
24-07-2023 5:57:07: Monitor Dimmed
24-07-2023 5:57:22: Monitor Power Off
24-07-2023 5:57:23: Backlight:5D-BA-C5-C4-00
24-07-2023 6:51:10: Monitor Power On
24-07-2023 6:51:10: AutoSetting for Offline
24-07-2023 6:51:10: Input: \\?\hid#vid_0b05&pid_19b6&mi_00&col01#8&1c08fbec&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
24-07-2023 6:51:10: Mode = 2 : OK
24-07-2023 6:51:10: Eco flag : 1
24-07-2023 6:51:10: Mux flag : -65536
24-07-2023 6:51:10: LQ134N1JW52:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
24-07-2023 6:51:10: BatteryLimit = 75 : OK
24-07-2023 6:51:10: USB-KB 0x19B6:5D-BD-01-FF-1E-0F
24-07-2023 6:51:11: USB-KB 64|32Dispositivo definido por el proveedor compatible con HID\\?\hid#vid_0b05&pid_19b6&mi_00&col03#8&1c08fbec&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
24-07-2023 6:51:11: Touchpad status:1
24-07-2023 6:51:11: Tablet: False Touchpad: True
24-07-2023 6:51:11: Auto:5D-BA-C5-C4-01
24-07-2023 6:51:11: PowerLimit A0 = 11 : OK
24-07-2023 6:51:11: PowerLimit A3 = 11 : OK
24-07-2023 6:51:41: WMI event 185
24-07-2023 6:51:44: WMI event 123
24-07-2023 6:51:44: Power Mode Changed
24-07-2023 6:51:44: AutoSetting for Online
24-07-2023 6:51:44: Input: \\?\hid#vid_0b05&pid_19b6&mi_00&col01#8&1c08fbec&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
24-07-2023 6:51:44: Mode = 2 : OK
24-07-2023 6:51:44: FanCPU = 32-3A-40-47-4C-51-53-55-00-0B-16-22-2F-42-53-63 : OK
24-07-2023 6:51:44: FanGPU = 32-38-3D-42-47-4D-51-55-00-0E-1B-2A-38-48-56-63 : OK
24-07-2023 6:51:44: Boost 6
24-07-2023 6:51:44: Running eco command 0
24-07-2023 6:51:44: USB-KB 0x19B6:5D-BD-01-FF-1E-0F
24-07-2023 6:51:44: USB-KB 64|32Dispositivo definido por el proveedor compatible con HID\\?\hid#vid_0b05&pid_19b6&mi_00&col03#8&1c08fbec&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
24-07-2023 6:51:44: Auto:5D-BA-C5-C4-01
24-07-2023 6:51:44: GPUEco = 0 : OK
24-07-2023 6:51:44: Touchpad status:1
24-07-2023 6:51:44: Tablet: False Touchpad: True
24-07-2023 6:51:44: Eco flag : 0
24-07-2023 6:51:44: Mux flag : -65536
24-07-2023 6:51:44: LQ134N1JW52:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
24-07-2023 6:51:47: NVIDIA GeForce GTX 1650 with Max-Q Design
24-07-2023 6:51:50: WMI event 186
24-07-2023 6:51:51: Key: 56
24-07-2023 6:51:51: Listener stopped
24-07-2023 6:51:51: Listener stopped
24-07-2023 6:51:55: WMI event 190
24-07-2023 6:51:56: WMI event 194
24-07-2023 6:52:01: GPU XGM = 1 : OK
24-07-2023 6:52:01: XGM 0x1970|300:5E-C5-00
24-07-2023 6:52:08: Key: 56
24-07-2023 6:52:16: XGM 0x1970|300:5E-D1-01-32-39-40-46-4B-4E-51-53-00-04-0E-1C-2E-3F-4F-64
24-07-2023 6:52:16: Eco flag : 0
24-07-2023 6:52:16: Mux flag : -65536
24-07-2023 6:52:17: Display configuration changed.
24-07-2023 6:52:17: Found external screen: C27FG7x:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_DISPLAYPORT_EXTERNAL
24-07-2023 6:52:18: Changed Lid Action to 0
24-07-2023 6:52:18: Engaging Clamshell Mode
24-07-2023 7:05:23: WMI event 186
24-07-2023 7:05:23: WMI event 190
24-07-2023 7:05:23: Display configuration changed.
24-07-2023 7:05:23: Changed Lid Action to 1
24-07-2023 7:05:23: Disengaging Clamshell Mode
24-07-2023 7:05:23: WMI event 194
24-07-2023 7:05:26: Key: 56
24-07-2023 7:05:41: GET GPU CLOCKS:NVAPI_API_NOT_INITIALIZED
24-07-2023 7:05:45: Mode = 0 : OK
24-07-2023 7:05:45: GET GPU CLOCKS:NVAPI_API_NOT_INITIALIZED
24-07-2023 7:05:45: SET GPU CLOCKS: 0, 0
24-07-2023 7:05:45: SET GPU CLOCKS: NVAPI_API_NOT_INITIALIZED
24-07-2023 7:05:45: Power mode: 00000000-0000-0000-0000-000000000000
24-07-2023 7:05:46: PowerLimit A0 = 27 : OK
24-07-2023 7:05:46: PowerLimit A3 = 27 : OK
24-07-2023 7:05:46: WMI event 186
24-07-2023 7:05:47: Key: 56
24-07-2023 7:05:47: Key: 56
24-07-2023 7:05:48: WMI event 190
24-07-2023 7:05:48: WMI event 194
24-07-2023 7:05:52: GPU XGM = 1 : OK
24-07-2023 7:05:52: XGM 0x1970|300:5E-C5-00
24-07-2023 7:05:58: GET GPU CLOCKS: 0, 0
24-07-2023 7:06:07: Eco flag : 0
24-07-2023 7:06:07: Mux flag : -65536
24-07-2023 7:06:08: Mode = 0 : OK
24-07-2023 7:06:08: XGM 0x1970|300:5E-D1-02
24-07-2023 7:06:08: GET GPU CLOCKS: 0, 0
24-07-2023 7:06:08: XGM 0x1970|300:5E-D1-01-14-3A-3D-40-48-4D-51-62-00-08-11-16-22-29-30-45
24-07-2023 7:06:08: FanCPU = 14-3A-3D-40-48-4D-51-62-00-08-11-16-22-29-30-45 : OK
24-07-2023 7:06:08: FanGPU = 14-3A-3D-40-48-4D-51-62-00-0C-16-1D-26-2D-34-4A : OK
24-07-2023 7:06:08: GET GPU CLOCKS: 0, 0
24-07-2023 7:06:08: WMI event 186
24-07-2023 7:06:08: WMI event 190
24-07-2023 7:06:08: Mode = 0 : OK
24-07-2023 7:06:09: GET GPU CLOCKS:NVAPI_NVIDIA_DEVICE_NOT_FOUND
24-07-2023 7:06:09: SET GPU CLOCKS: 0, 0
24-07-2023 7:06:09: SET GPU CLOCKS: NVAPI_API_NOT_INITIALIZED
24-07-2023 7:06:09: GET GPU CLOCKS:NVAPI_API_NOT_INITIALIZED
24-07-2023 7:06:09: WMI event 194
24-07-2023 7:06:09: PowerLimit A0 = 27 : OK
24-07-2023 7:06:09: PowerLimit A3 = 27 : OK
24-07-2023 7:06:10: PowerLimit A0 = 27 : OK
24-07-2023 7:06:10: PowerLimit A3 = 27 : OK
24-07-2023 7:06:21: WMI event 186
24-07-2023 7:06:23: WMI event 186
24-07-2023 7:07:00: Key: 56
24-07-2023 7:07:01: Key: 56
24-07-2023 7:07:02: WMI event 186
24-07-2023 7:07:03: Key: 56
24-07-2023 7:07:05: WMI event 190
24-07-2023 7:07:05: WMI event 194
24-07-2023 7:07:08: GPU XGM = 1 : OK
24-07-2023 7:07:08: XGM 0x1970|300:5E-C5-00
24-07-2023 7:07:09: Mode = 1 : OK
24-07-2023 7:07:09: Power mode: ded574b5-45a0-4f42-8737-46345c09c238
24-07-2023 7:07:09: Boost 4
24-07-2023 7:07:17: GET GPU CLOCKS:NVAPI_API_NOT_INITIALIZED
24-07-2023 7:07:22: Mode = 1 : OK
24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-02
24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-01-14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A
24-07-2023 7:07:22: FanCPU = 14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A : OK
24-07-2023 7:07:22: FanGPU = 14-3F-44-48-4C-50-54-62-16-1F-26-2D-39-47-55-5F : OK
24-07-2023 7:07:22: GET GPU CLOCKS: 0, 0
24-07-2023 7:07:23: Mode = 1 : OK
24-07-2023 7:07:23: XGM 0x1970|300:5E-D1-02
24-07-2023 7:07:23: GET GPU CLOCKS: 0, 0
24-07-2023 7:07:23: PowerLimit Fix A0 = 80 : OK
24-07-2023 7:07:23: PowerLimit Fix A3 = 80 : OK
24-07-2023 7:07:23: Eco flag : 0
24-07-2023 7:07:23: Mux flag : -65536
24-07-2023 7:07:33: Key: 56
24-07-2023 7:07:35: Key: 56
24-07-2023 7:07:54: Display configuration changed.
24-07-2023 7:07:54: Found external screen: C27FG7x:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_DISPLAYPORT_EXTERNAL
24-07-2023 7:07:54: Changed Lid Action to 0
24-07-2023 7:07:54: Engaging Clamshell Mode
24-07-2023 7:12:25: Key: 56
24-07-2023 7:12:28: Mode = 0 : OK
24-07-2023 7:12:28: XGM 0x1970|300:5E-D1-02
24-07-2023 7:12:28: GET GPU CLOCKS: 0, 0
24-07-2023 7:12:28: Power mode: 00000000-0000-0000-0000-000000000000
24-07-2023 7:12:28: Boost 6
24-07-2023 7:12:29: PowerLimit A0 = 27 : OK
24-07-2023 7:12:29: PowerLimit A3 = 27 : OK
24-07-2023 7:12:30: GET GPU CLOCKS: 0, 0
24-07-2023 7:12:33: Key: 56
24-07-2023 7:12:33: Key: 56
24-07-2023 7:12:34: Key: 56
24-07-2023 7:12:35: Key: 56
24-07-2023 7:12:35: Key: 56
24-07-2023 7:12:44: Key: 56
24-07-2023 7:12:45: GET GPU CLOCKS: 0, 0
24-07-2023 7:32:12: Monitor Dimmed
24-07-2023 7:32:27: Monitor Power Off
24-07-2023 7:32:28: Backlight:5D-BA-C5-C4-00
24-07-2023 7:33:34: Monitor Power On
24-07-2023 7:33:34: AutoSetting for Online
24-07-2023 7:33:34: Input: \\?\hid#vid_0b05&pid_19b6&mi_00&col01#8&1c08fbec&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
24-07-2023 7:33:34: Mode = 0 : OK
24-07-2023 7:33:34: XGM 0x1970|300:5E-D1-02
24-07-2023 7:33:34: GET GPU CLOCKS: 0, 0
24-07-2023 7:33:34: Eco flag : 0
24-07-2023 7:33:34: Mux flag : -65536
24-07-2023 7:33:34: C27FG7x:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_DISPLAYPORT_EXTERNAL: \\.\DISPLAY18
24-07-2023 7:33:34: BatteryLimit = 75 : OK
24-07-2023 7:33:34: USB-KB 0x19B6:5D-BD-01-FF-1E-0F
24-07-2023 7:33:34: USB-KB 64|32Dispositivo definido por el proveedor compatible con HID\\?\hid#vid_0b05&pid_19b6&mi_00&col03#8&1c08fbec&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
24-07-2023 7:33:34: XGM 0x1970|300:5E-C5-00
24-07-2023 7:33:34: Touchpad status:1
24-07-2023 7:33:34: Tablet: False Touchpad: True
24-07-2023 7:33:34: Auto:5D-BA-C5-C4-01
24-07-2023 7:33:35: PowerLimit A0 = 27 : OK
24-07-2023 7:33:35: PowerLimit A3 = 27 : OK
24-07-2023 7:34:20: Listener stopped
24-07-2023 7:35:50: Key: 56
24-07-2023 7:35:51: GET GPU CLOCKS: 0, 0
24-07-2023 7:35:54: Key: 56
24-07-2023 7:36:07: Key: 56
24-07-2023 7:36:08: GET GPU CLOCKS: 0, 0
24-07-2023 7:36:18: Key: 56
24-07-2023 7:40:22: Key: 56
24-07-2023 7:40:25: Key: 56

**Desktop (please complete the following information):**
 - OS: Windows 11
 - ROG Flow X13 gv301qh
 - XG Mobile with RTX 3080

**Asus software**
- Every ASUS software or service I can think of is uninstalled or disabled

**Additional context**
I don't have any other software to control fan curves, but I do have MSI Afterburner installed and active.

**Comment by seerge** [comment]
@Plumbio hello,

From log it's clear that app actually sends a custom fan curve to XGM when you select your mode (turbo in your case)

```
24-07-2023 7:07:22: Mode = 1 : OK
24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-02
24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-01-14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A
24-07-2023 7:07:22: FanCPU = 14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A : OK
24-07-2023 7:07:22: FanGPU = 14-3F-44-48-4C-50-54-62-16-1F-26-2D-39-47-55-5F : OK
```

This is XGM fan curve ``24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-01-14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A``

Does custom fan curves generally work for your 3080 ? What happens you click apply again under fans+power?

**Comment by seerge** [comment]
@Plumbio since i didn't change anything related to XGM (and in general to fan curves) for like last 2 months or so, it looks more like some issue / hiccup with your XGM

I remember that in this topic (irrelevant to you) https://github.com/seerge/g-helper/issues/729 issue-poster was doing hard reset for XGM, mb that's something you should try ?

**Comment by Plumbio** [comment]
> @Plumbio hello,
> 
> From log it's clear that app actually sends a custom fan curve to XGM when you select your mode (turbo in your case)
> 
> ```
> 24-07-2023 7:07:22: Mode = 1 : OK
> 24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-02
> 24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-01-14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A
> 24-07-2023 7:07:22: FanCPU = 14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A : OK
> 24-07-2023 7:07:22: FanGPU = 14-3F-44-48-4C-50-54-62-16-1F-26-2D-39-47-55-5F : OK
> ```
> 
> This is XGM fan curve `24-07-2023 7:07:22: XGM 0x1970|300:5E-D1-01-14-3F-44-48-4C-50-54-62-11-1A-22-29-34-43-51-5A`
> 
> Does custom fan curves generally work for your 3080 ? What happens you click apply again under fans+power?

If I deactivate the fan curve, the fan comes back to it's default curve and starts to spin, but if I activate the custom curve again, it locks to it's current speed, so it seems like the custom curve is preventing the fan's RPM from changing.

It happens with any energy plan that has a custom fan curve activated, and if I connect the XGM with a fan curve already enabled, I assume that the XGM fan gets locked at 0 RPM and that's why it's like the fan were "deactivated".



> @Plumbio since i didn't change anything related to XGM (and in general to fan curves) for like last 2 months or so, it looks more like some issue / hiccup with your XGM
> 
> I remember that in this topic (irrelevant to you) #729 issue-poster was doing hard reset for XGM, mb that's something you should try ?

The first time this happened, I didn't know what was wrong, the XGM were disconnecting by itself and tried a bunch of things until I touched it and felt how hot it was, so I connected it, disabled the custom fan curve and it went full rpm, BUT one of the thing I tryed before was reconnecting the plug a couple times and that did nothing.

Also some time ago I had an issue where my XGM wasn't able to go beyond 115W, and the solution was to do a clean install of the nvidia drivers with the XGM unplugged. For some reason, if you install the nvidia drivers while having the xgm connected to your laptop, it's gonna be locked at 115W.
What I'm trying to say is that my xgm has it's drivers correctly installed and hard resetting doesn't work (if I really understood what "hard resetting" is) and it also has it's BIOS (laptop and xgm) up to date.

I have armoury crate control interface disabled in the BIOS, maybe that could be the problem?

PD: to be clear, custom fan curves are messing only with my xgm, but they work with no problem with the laptop fans.

**Comment by seerge** [comment]
We can ask @Wolper123 how did he do hardware reset of XGM mb? :)

**Comment by seerge** [comment]
@Wolper321 is it power button on XG Mobile, right ? @Plumbio can you try that as well?

**Comment by seerge** [comment]
@Wolper321 Max RPM that chart shows is just a UI (i.e. cosmetic). As in reality all fan curves are set in % (from 0 to 100). So it's same for all cases and never changed.

I have added a tweak and now XGM fan will show 7200 RPM (if it's a known max) 
You can check it here [GHelper.zip](https://github.com/seerge/g-helper/files/12163679/GHelper.zip)


@Plumbio for your situation - i don't know why your fan doesn't react on custom fan curve (as in nothing has changed in the app related to that + log shows actual data that is being set to fan, it's all ok). And I can't really help here.

Did you had a chance to test it with AC?

**Comment by seerge** [comment]
@Plumbio hello, thanks for a detailed "investigation" :) 

> After that I installed AC, made a fan curve with thar software and it worked, then disabled manual mode, tried with the an curve on g-helper and it also worked, almost no problems this time. 

It could be that there is some "initialization" command that AC can send to XGM in order to  wake it up. So after command is sent once - everything will work, also from g-helper. It's a similar command for Aura (backlight), that I know.

Can you check this build - it will send aura command to XGM, prior to setting fan curve there, mb it will help
**[GHelper.zip](https://github.com/seerge/g-helper/files/12171770/GHelper.zip)**

> The only thing not working was the RPM monitoring in g-helper, it was showing 0 rpm permanently.

G-Helper never could read XGM fans. What it shows under GPU fan is a sensor from BIOS that is supposed to be internal GPU fan (is AC able to show XGM fans specifically?).

> But I noticed that when I turn on my laptop and connect my xgm, g-helper doesn't show the xgm's gpu temps until I disable the custom curve and then g-helper start to show the external gpu's temps instead of the internal one, but if I reenable the fan curve, 

This is not related to custom curve :) g-helper gets GPU temps from Nvidia / AMD drivers. And driver also can report temp for XGM obviously. You just triggered app to reconnect to driver when opened fans+power screen.

**Comment by Plumbio** [comment]
Sorry, it didn't fix the problem, but I found out something new.

If I restart my laptop while the xgm is connected, the fan curve will apply, this works if it is either with a custom curve already applied or with the deffault curve and then applying the custom one after the restart. But if I disconnect the xgm and reconnect it, the problem comes back.

> G-Helper never could read XGM fans. What it shows under GPU fan is a sensor from BIOS that is supposed to be internal GPU fan (is AC able to show XGM fans specifically?).

I think AC doesn't show the xgm's fan rpm. At some moment the fan was full blowing, but AC and G-helper were showing like 3100 rpm.

Also, I assume this comment will be out of place, but the button to close the app in spanish says "Quitar", which means "take off" or "remove". I think "Cerrar" would be more accurate.

And thanks again for your help, your app is really cool, I wouldn't go back to AC even if my problem isn't fixed.

**Comment by seerge** [comment]
@Plumbio then it could be that there is some other payload that needs to be sent to XGM :) I don't know it., cause I don't have device itself.

There is a way to try to catch it tho, cause all communication between app/windows and XGM is actually over USB . So you can install app https://www.hhdsoftware.com/device-monitoring-studio and then just intercept all packages that is sent from Armoury Crate (starting from it's launch) to a device with id  0x1970 (or just called XG Mobile)

Let me know if you want to try that :) otherwise, i'm afraid there is nothing i can do here.

**Comment by Wolper321** [comment]
> G-Helper never could read XGM fans. What it shows under GPU fan is a sensor from BIOS that is supposed to be an internal GPU fan (is AC able to show XGM fans specifically?).
XGM fan is a bit special. AMD couldn't read it but Armoury Crate does and read it in a very precise manner. 
![XG_FANS](https://github.com/seerge/g-helper/assets/12784148/46f70f85-0b21-43a7-9468-ed9842f140fe)
In my case, I don't have dgpu in the laptop so the fan is called  "CPU aux" instead of "GPU" fan.

**Comment by Wolper321** [comment]
> @Plumbio then it could be that there is some other payload that needs to be sent to XGM :) I don't know it., cause I don't have device itself.
> 
> There is a way to try to catch it tho, cause all communication between app/windows and XGM is actually over USB . So you can install app https://www.hhdsoftware.com/device-monitoring-studio and then just intercept all packages that is sent from Armoury Crate (starting from it's launch) to a device with id 0x1970 (or just called XG Mobile)
> 
> Let me know if you want to try that :) otherwise, i'm afraid there is nothing i can do here.
> 
> P.S. As for Quitar - it was already a discussion between 2 Spanish natives [#225 (comment)](https://github.com/seerge/g-helper/discussions/225#discussioncomment-5839861) :)

I am actually willing to investigate more about it but I only have curiosity. Technical skills are not included :D 

Here is the pic with the devices attached:
![image](https://github.com/seerge/g-helper/assets/12784148/540cec20-fbcc-43e8-b200-b5357b3b0939)

I believe this ITE Device(8910) is responsible:

Connection Information
 Port: 3
 Speed: Full Speed
 Device address: 7
 Open pipes: 1
 Connection status: Device connected

Device Descriptor
 USB version: 2.0
 Device class: 0x0 - (Defined at Interface level)
 Device subclass: 0x0 - Unknown
 Device protocol: 0x0 - Unknown
 Control pipe max size: 64 bytes
 Vendor ID: 0xb05 (ASUSTek Computer, Inc.)
 Product ID: 0x1970 (Unknown)
 Product version: 0.3
 Manufacturer: ITE Tech. Inc.
 Product: ITE Device(8910)
 Serial Number: Not specified
 Configurations: 1

**Comment by Plumbio** [comment]
Hi, I did the monitoring, but I think it might not be usefull.
The last time I did some testing with AC, I had to install it, and thus, had to restart my laptop. That made me think that everything was ok with AC, but now that I was creating the log files, I reinstalled AC again and recreated the scenarios with both AC and G-helper, and the problem happened with both programs.
So.... it seems that this is a problem with my laptop and has nothing to do with g-helper.

Could it be a driver problem?
I have 535.98 installed, maybe I should try updating or installing the one from the asus support page.

@Wolper321 could you tell me if you use the drivers from asus support or the latest ones from nvidia please?

I'm attaching logs from after the laptop restart (when custom curves do work) and after the xgm reconnection (curves stop working) in case they are usefull of something.

Oh, and also, I didn't notice that there was more to be seen on the AC fan speed tab, so now I can confirm that it gets locked to its current RPM when I enable a custom fan curve, and the highest speed I saw was 7784 rpm (I'm mentioning that because I read that you set the 100% speed rpm value manually on G-helper).

And sorry @seerge , as it seems that g-helper was never the problem, I might just have been wasting your time.
[xgm monitoring log.txt](https://github.com/seerge/g-helper/files/12206144/xgm.monitoring.log.txt)
[xgm monitoring log 2.txt](https://github.com/seerge/g-helper/files/12206145/xgm.monitoring.log.2.txt)

**Comment by seerge** [comment]
@Plumbio thanks for logging :)

I have checked logs, and actually i don't seem to find any known setting data here at all (fan curve / red-led for example has to contain 0x5e symbol in the beginning)

But after reading that your initial problem seem to happen with AC as well, it indeed could be some hiccup with your device only then.

I don't think this is related to a driver, as fans are controlled by firmware (i.e. bios on laptop, or some "internal firmware" on XGM). Does XGM offer firmware updates in general? mb it makes sense to re-flush it (with just latest version of it) ?

**Comment by Mikealll** [comment]
Sorry, did not have time to read the entire thread. However, I have noticed a similar issue but it is easily resolved. Custom fan curves will not work on xgm (fan will not turn on.) If the XGM has just been connected to the laptop, but after hibernating or shutting down it will work as intended.

**Comment by Plumbio** [comment]
@Mikealll I didn't try hibernating, but restarting the laptop "fixes" the problem. As you said, it happens only after connecting the xgm.

@seerge I updated the xgm firmware, but there's no change, BUT I found out simething.
If I switch the GPU to Eco mode and then go back to Optimized while the XGM is connected, custom curves work again (during that session). So it seems like the external gpu needs to be disabled and reenabled to "fix" the problem.
But it's really weird that my xgm's gpu needs to be enabled, disabled and enabled again for the custom curves to work.

I'm attaching my g-helper logs in case there is new info.
[log.txt](https://github.com/seerge/g-helper/files/12223094/log.txt)

**Comment by seerge** [comment]
@Plumbio clear :) well re-enabling dGPU as "default" action is probably too much for all devices.

It's clear that on this stage I can't do anything. If (eventually) I will manage to get my hands on XGM I will catch whole USB communications there and check if armoury does anything extra to what g-helper already doe.

I'm closing this as I can't do anything more atm :)

Thanks for testing.
